sheet_to_triples/tests/test_rdf.py

```diff
@@ -0,0 +1,242 @@
+        ]
+        self.assertIsInstance(term, rdflib.paths.SequencePath)
+        self.assertEqual(
+            [str(p) for p in term.args],
+            expected
+        )
+
+    def test_from_identifier_prefix_http(self):
+        term = rdf.from_identifier('http://test.test')
+        self.assertIsInstance(term, rdflib.term.URIRef)
+        self.assertEqual(str(term), 'http://test.test')
+
+    def test_from_identifier_literal(self):
+        self.assertEqual(
+            rdf.from_identifier('test'),
+            rdflib.Literal('test')
+        )
+
+    def test_relates_geo_name_true(self):
+        for path in ('atGeoPoint', 'atGeoPoly', 'name'):
+            term = {'pred': 'http://visual-meaning.com/rdf/' + path}
+            with self.subTest(pred=term['pred']):
+                self.assertTrue(rdf.relates_geo_name(term))
+
+    def test_relates_geo_name_false(self):
+        for path in ('http://visual-meaning.com/rdf/test', 'test/atGeoPoint'):
+            term = {'pred': path}
+            with self.subTest(pred=term['pred']):
+                self.assertFalse(rdf.relates_geo_name(term))
+
+    def test_relates_issue_true(self):
+        self.assertTrue(
+            rdf.relates_issue({'subj': 'http://visual-meaning.com/rdf/test'})
+        )
+
+    def test_relates_issue_false(self):
+        self.assertFalse(
+            rdf.relates_issue(
+                {'subj': 'http://visual-meaning.com/rdf/issues/'},
+            )
+        )
+
+    def test_purge_terms(self):
+        model = {
+            'terms': [
+                {'obj': 'http://visual-meaning.com/rdf/test'},
+                {'obj': 'http://visual-meaning.com/rdf/whitespace   '},
+                {'obj': 'test'}
+            ]
+        }
+
+        def retain(x):
+            return x['obj'].startswith('http://visual-meaning.com/rdf/')
+
+        rdf.purge_terms(model, retain, False)
+
+        expected = {
+            'terms': [
+                {'obj': 'http://visual-meaning.com/rdf/test'},
+                {'obj': 'http://visual-meaning.com/rdf/whitespace '},
+            ]
+        }
+        self.assertEqual(model, expected)
+
+    def test_graph_from_model(self):
+        model = {
+            'terms': [
+                {
+                    'subj': 'test_subj',
+                    'pred': 'test_pred',
+                    'obj': 'test_obj',
+                }
+            ]
+        }
+        graph = rdf.graph_from_model(model)
+
+        instance_values = [
+            (next(graph.subjects()), rdflib.term.URIRef, 'test_subj'),
+            (next(graph.predicates()), rdflib.term.URIRef, 'test_pred'),
+            (next(graph.objects()), rdflib.term.Literal, 'test_obj'),
+        ]
+        for test_value, instance, expected_value in instance_values:
+            with self.subTest(value=expected_value):
+                self.assertIsInstance(test_value, instance)
+                self.assertEqual(str(test_value), expected_value)
```

Comment:
This validation is pretty hard to read. Would be easier to just make a ttl string which encodes all these details in one simple string compare?